src/lightning/app/runners/cloud.py

```diff
@@ -529,16 +540,29 @@ def _resolve_cloudspace_name(
     ) -> str:
         """If there are existing cloudspaces but not on the cluster - choose a randomised name."""
         if len(existing_cloudspaces) > 0 and existing_cloudspace is None:
-            letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
-
             name_exists = True
             while name_exists:
-                random_name = cloudspace_name + "-" + "".join(random.sample(letters, 4))
+                random_name = cloudspace_name + "-" + "".join(random.sample(string.ascii_letters, 4))
                 name_exists = any([app.name == random_name for app in existing_cloudspaces])
 
             cloudspace_name = random_name
         return cloudspace_name
 
+    def _resolve_run_name(
+        self,
+        name: str,
+        existing_instances: List[Externalv1LightningappInstance],
+    ) -> str:
+        """If there are existing instances with the same name - choose a randomised name."""
+        if len(existing_instances) > 0:
+            name_exists = True
+            while name_exists:
```

Comment:
rather a foe loot with exit if you tried to many times

Reply:
We will be refactoring this stuff quite a bit soon I think. Leaving for then